rudder-core/src/main/scala/com/normation/rudder/db/Doobie.scala

```diff
@@ -179,6 +180,20 @@ object Doobie {
     )
   }
 
+  implicit val DataSourceComposite: Composite[RudderDataSource] = {
+    import com.normation.rudder.repository.json.CompleteJson._
+    import com.normation.rudder.datasources.DataSourceJsonSerializer._
+    import net.liftweb.json.compactRender
+    import net.liftweb.json.parse
+    Composite[(DataSourceId,String)].xmap(
+        tuple => extractDataSource(tuple._1,parse(tuple._2)) match {
+          case Full(s) => s
+          case eb  => throw new RuntimeException("oops")
```

Comment:
Could you change the message to log what is the cause of the error ("error when deserialising blabla") and perhaps even the faulty json (not sure if it is already logged in extractDataSource)